core/templates/domain/blog/blog-post-editor-backend-api.service.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to send changes to a blog post to the backend.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { BlogPostBackendDict, BlogPostData, BlogPostObjectFactory } from 'domain/blog/BlogPostObjectFactory';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { HttpClient } from '@angular/common/http';
+import { BlogDashboardPageConstants } from 'pages/blog-dashboard-page/blog-dashboard-page.constants';
+import { BlogPostChangeDict } from 'domain/blog/blog-post-update.service';
+
+interface ImageData {
+  filename: string,
+  imageBlob: Blob
+}
+
+interface BlogPostUpdateBackendDict {
+  'blog_post': BlogPostBackendDict;
+}
+
+interface BlogPostUpdatedData {
+  blogPostDict: BlogPostData
+}
+
+interface DeleteBlogPostBackendResponse {
+  status: number;
+}
+
+interface BlogPostEditorBackendResponse {
+  'blog_post_dict': BlogPostBackendDict;
+  'username': string;
+  'profile_picture_data_url': string;
+  'max_no_of_tags': number;
+  'list_of_default_tags': string[];
+}
+
+export interface BlogPostEditorData {
+  blogPostDict: BlogPostData;
+  username: string;
+  profilePictureDataUrl: string;
+  maxNumOfTags: number;
+  listOfDefaulTags: string[];
+}
```

Comment:
Please move them to a separate typings file.

Reply:
Done.

Reply:
Reverted changes.